### PR TITLE
Add async support to fetch pages

### PR DESCRIPTION
**Pull request by dbaxa**

### CODE REVIEW

**Comment by hdost** [comment]
It looks like some of these commits are from master and then others are not. Is there a way to distill this down to its necessary parts?

**Comment by dbaxa** [comment]
@hdost if we merge #518, then it will become clear which commits are for master and which are not.

**Comment by hdost** [comment]
The point I am trying to bring across is  was also mentioned by https://github.com/pycontribs/jira/pull/521#pullrequestreview-100394105
The way that you do PRs is to have the commits which are relevant to the PR. #518 should be or not be merged on it's own merits. And this one should be separate from that. We want small changes going in at a time so its understandable what the purpose of merges does. 

fa23162  has nothing to do with b32a9ab  which has nothing to do with merging from master. Try to understand where we are coming from. I do appreciate the effort being made, but for us to make this more maintainable and understandable smaller changes are necessary.

**Comment by Stealthii** [comment]
Yup. If you're going to merge into `develop`, you should branch off `develop`. I suggest rebasing your PR branch from there and tidying up the commits so that this change is unique.

**Comment by dbaxa** [comment]
Hi @hdost & @Stealthii , I understand your request but you seem to have missed what I have done by opening these prs. First off I can't just do what you have suggested in this case, no matter how much i wanted to do so, because the `develop` branch was left with failing tests it seems since https://github.com/pycontribs/jira/commit/3927ad717d2e8b40204650a13e17a3d6018e47a4 ... which was in November 2016. The point of this pr is to implement a feature and in order to show that my change does not break this project I wanted the tests to be green first. So I spent time fixing making sure what I branched off actually passed tests. After #518 and #519 are merged in some fashion I can rebase / update this pull request appropriately. Or I can open a new one.

**Comment by dbaxa** [comment]
I have rebased this off the develop branch.

File: jira/utils/__init__.py [hunk 1]
```diff
@@ -77,8 +76,10 @@ def threaded_requests(requests):
 
 def json_loads(r):
```
Comment (dbaxa) [inline jira/utils/__init__.py]: @hdost I can split this change out into another pull request if that is desired.

**Comment by dbaxa** [comment]
@ssbarnea I want to merge this soon.

**Comment by ssbarnea** [comment]
Due to changes made to develop and master branches I have to close this and start this on different branch.

See https://github.com/pycontribs/jira/pull/538 and review it. If I missed some of your changes please try to create a PR against this branch.

**Pull request closed without merging**